lib/Dialect/TritonGPU/Transforms/Pipeliner/PipeliningUtility.cpp

```diff
@@ -574,7 +577,12 @@ mlir::triton::getMultiBufferedType(ttg::MemDescType memDescType,
                                    int32_t depth) {
   auto shape = memDescType.getShape();
   SmallVector<int64_t> bufferShape(shape.begin(), shape.end());
-  bufferShape.insert(bufferShape.begin(), depth);
+  if (!isa<nvidia_gpu::TensorMemoryScalesEncodingAttr>(
+          memDescType.getEncoding())) {
+    bufferShape.insert(bufferShape.begin(), depth);
+  } else {
+    assert(depth == 1 && "Scales don't currently support multibuffering");
+  }
```

Comment:
why do we call this function for scales if we don't support multibuffering for them?

Reply:
we need to create a mutable type even if shape remains the same, e.g. `%buf = tmem_alloc %src` -> `%buf = tmem_alloc; tmem_store %src, %buf`; created a  separate aref helper function in nvws namespace to handle such multi-buffering.